Plausible/DeriveArbitrary.lean

```diff
@@ -0,0 +1,281 @@
+    let mut argNamesAndTypes := #[]
+    for arg in args.toList do
+      let localDecl ← arg.fvarId!.getDecl
+      let mut argName := localDecl.userName
+      -- Check if the name has macro scopes
+      -- If so, remove them so that we can produce a user-accessible name
+      -- (where macro scopes don't appear in the name)
+      if argName.hasMacroScopes then
+        argName := Name.eraseMacroScopes argName
+      argNamesAndTypes := Array.push argNamesAndTypes (argName, localDecl.type)
+
+    return argNamesAndTypes
+
+/-- Produces an instance of the `ArbitrarySized` typeclass for an inductive type
+    whose name is given by `targetTypeName`.
+
+    (Note: the main logic for determining the structure of the derived generator
+    is contained in this function.) -/
+def mkArbitrarySizedInstance (targetTypeName : Name) : CommandElabM (TSyntax `command) := do
+  -- Obtain Lean's `InductiveVal` data structure, which contains metadata about
+  -- the type corresponding to `targetTypeName`
+  let inductiveVal ← getConstInfoInduct targetTypeName
+
+  -- Fetch the ambient local context, which we need to produce user-accessible fresh names
+  let localCtx ← liftTermElabM $ getLCtx
+
+  -- Produce a fresh name for the `size` argument for the lambda
+  -- at the end of the generator function, as well as the `aux_arb` inner helper function
+  let freshSizeIdent := mkIdent $ localCtx.getUnusedName `size
+  let freshSize' := mkIdent $ localCtx.getUnusedName `size'
+
+  let mut nonRecursiveGenerators := #[]
+  let mut recursiveGenerators := #[]
+  for ctorName in inductiveVal.ctors do
+    let ctorIdent := mkIdent ctorName
+
+    let ctorArgNamesTypes ← liftTermElabM $ getCtorArgsNamesAndTypes ctorName
+    let (ctorArgNames, ctorArgTypes) := Array.unzip ctorArgNamesTypes
+
+    /- Produce fresh names for each of the constructor's arguments.
+      Producing fresh names is necessary in order to handle
+      constructors expressed using the following syntax:
+      ```
+      inductive Foo
+        | C : T1 → ... → Tn
+      ```
+      in which all the arguments to the constructor `C` don't have explicit names.
+
+      (Note: we use `genFreshNames` instead of `LocalContext.getUnusedName` here
+      to ensure that when there are multiple arguments,
+      every single argument gets a fresh name. All of the fresh names are added
+      to the `LocalContext` later in this function, ensuring that the `LocalContext`
+      remains updated.)
+    -/
+    let freshArgNames := genFreshNames (existingNames := ctorArgNames) (namePrefixes := ctorArgNames)
+    let freshArgNamesTypes := Array.zip freshArgNames ctorArgTypes
+    let freshArgIdents := Lean.mkIdent <$> freshArgNames
+    let freshArgIdentsTypes := Array.zip freshArgIdents ctorArgTypes
+
+    if ctorArgNamesTypes.isEmpty then
+      -- Constructor is nullary, we can just use a generator of the form `pure ...`
+      let pureGen ← `($pureFn $ctorIdent)
```

Comment:
It's much easier to use double backticks to pre-resolve the names. There's no need for any of these `Ident` global constants in `Plausible.Idents`.
```suggestion
      let pureGen ← ``(pure $ctorIdent)
```